crates/pallet-domains/src/lib.rs

```diff
@@ -1570,29 +1568,26 @@ impl<T: Config> Pallet<T> {
             }
             FraudProof::InvalidExtrinsicsRoot(proof) => {
```

Comment:
How about moving the code of getting `block_randomness`, `domain_timestamp_extrinsic`, and `maybe_domain_set_code_extrinsic` to `verify_invalid_domain_extrinsics_root_fraud_proof`, to keep `pallet-domains` from bloating.

Reply:
I was planning to refactor the frud proof verification in pure refactoring PR. I will handle this suggestion then 👍🏼